experiments/AtmosLES/baroclinicwave.jl

```diff
@@ -0,0 +1,332 @@
+  eta = max(eta,FT(0))
+  ### η = p/p_s
+  logeta = log(eta)
+  T=FT(temp)
+  press = p00*eta
+  theta = T *(p00/press)^(R_gas/cp)
+  rho = press/(R_gas*T)
+  thetaref = T_ref * (1 - gamma_lapse*z/T0)^(-gravity/(cp*gamma_lapse))
+  rhoref = p00/(T0*R_gas) * (1 - gamma_lapse*z/T0)^(gravity/(R_gas*gamma_lapse) - 1)
+
+  ### Balanced Flow
+  u = -u0*(sinpi(y/Ly))^2  * logeta * exp(-logeta*logeta/b2)
+
+  ### Perturbation of the balanced flow
+  rc2 = (x-xc)^2 + (y-yc)^2
+  du = up*exp(-rc2/Lp2)
+    
+  ### Primitive variables
+  u⃗ = SVector{3,FT}(u+du,0,0)
+  e_kin = FT(1/2)*sum(abs2.(u⃗))
+  e_pot = gravity * z
+
+  ### Assign state variables for initial condition
+  state.ρ = rho
+  state.ρu = rho .* u⃗
+  state.ρe = rho * total_energy(param_set, e_kin, e_pot, T)
+end
+
+function config_baroclinicwave(FT, N, resolution, xmax, ymax, zmax)
+
+    ics = init_baroclinicwave!     # Initial conditions
+    C_smag = FT(0.30)     # Smagorinsky coefficient
+    
+    # Assemble source components
+    source = (
+        Gravity(),
+    )
+
+    # Choose default IMEX solver
+    ode_solver_type = ClimateMachine.IMEXSolverType()
+
+    # Assemble model components
+    model = AtmosModel{FT}(
+        AtmosLESConfigType,
+        param_set;
+        turbulence = SmagorinskyLilly{FT}(C_smag),
+        moisture = DryModel(),
+        source = source,
+        boundarycondition = (
+            AtmosBC(),
+            AtmosBC(),
+        ),
+        init_state_conservative = ics,
+    )
+
+    # Assemble configuration
+    config = ClimateMachine.AtmosLESConfiguration(
+        "Baroclinic Wave",
+        N,
+        resolution,
+        xmax,
+        ymax,
+        zmax,
+        param_set,
+        init_baroclinicwave!,
+        solver_type = ode_solver_type,
+        model = model,
+    )
+    return config
+end
+
+function config_diagnostics(driver_config, FT)
+    
+    boundaries = [
+        FT(0.0) FT(0.0) FT(0.0)
+        FT(4e7) FT(6e6) FT(30e3)
+    ]
+    resolution = (FT(100e3), FT(75e3), FT(1250)) 
+    interpol = ClimateMachine.InterpolationConfiguration(
+        driver_config,
+        boundaries,
+        resolution,
+    )
+
+    default_dgngrp = setup_atmos_default_diagnostics(
+        AtmosLESConfigType(),
+        "1shours",
+        driver_config.name,
+    )
+    core_dgngrp = setup_atmos_core_diagnostics(
+        AtmosLESConfigType(),
+        "1shours",
+        driver_config.name,
+    )
+    state_dgngrp = setup_dump_state_diagnostics(
+        AtmosLESConfigType(),
+        "1shours",
+        driver_config.name,
+        interpol = interpol,
+    )
+    aux_dgngrp = setup_dump_aux_diagnostics(
+        AtmosLESConfigType(),
+        "1shours",
+        driver_config.name,
+        interpol = interpol,
+    )
+    return ClimateMachine.DiagnosticsConfiguration([
+        default_dgngrp,
+    ])
+end
+
+function main()
+    FT = Float32
+
+    # DG polynomial order
+    N = 4
+    # Domain resolution and size
+    Δx = FT(1e5) 
```

Comment:
For a faster run in testing mode, Dx could be increased to 2e5. Based on the test shown with numa in Marras, Nazarov, Giraldo 2015, this problem should stay stable for up to Dx = 4e5. Worth testing.